--- org.springframework.web.servlet/pom.xml	2012-05-28 22:22:09.863940376 +0200
+++ org.springframework.web.servlet/pom.xml-gil	2012-05-28 22:26:27.471941775 +0200
@@ -128,6 +128,12 @@
       <optional>true</optional>
     </dependency>
     <dependency>
+      <groupId>org.apache.tiles</groupId>
+      <artifactId>tiles-el</artifactId>
+      <version>2.1.2</version>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
       <groupId>velocity</groupId>
       <artifactId>velocity</artifactId>
       <version>1.5</version>
