--- org.springframework.orm/pom.xml	2013-01-23 14:58:18.000000000 +0100
+++ org.springframework.orm/pom.xml-gil	2013-10-12 15:07:36.949803877 +0200
@@ -35,39 +35,69 @@
     </dependency>
     <!-- common logging dependency is in spring-core -->
     <dependency>
-      <groupId>org.apache.ibatis</groupId>
-      <artifactId>ibatis-sqlmap</artifactId>
-      <version>2.3.4.726</version>
+      <groupId>org.apache.openjpa</groupId>
+      <artifactId>openjpa-lib</artifactId>
+      <version>2.2.1</version>
+      <optional>true</optional>
+     </dependency>
+    <dependency>
+      <groupId>org.apache.openjpa</groupId>
+      <artifactId>openjpa-kernel</artifactId>
+      <version>2.2.1</version>
       <optional>true</optional>
     </dependency>
     <dependency>
       <groupId>org.apache.openjpa</groupId>
-      <artifactId>openjpa</artifactId>
-      <version>1.1.0</version>
+      <artifactId>openjpa-jdbc</artifactId>
+      <version>2.2.1</version>
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>org.eclipse.persistence</groupId>
-      <artifactId>org.eclipse.persistence.core</artifactId>
-      <version>1.0.1</version>
+      <groupId>org.apache.openjpa</groupId>
+      <artifactId>openjpa-persistence</artifactId>
+      <version>2.2.1</version>
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>org.eclipse.persistence</groupId>
-      <artifactId>org.eclipse.persistence.jpa</artifactId>
-      <version>1.0.1</version>
+      <groupId>org.apache.openjpa</groupId>
+      <artifactId>openjpa-persistence-jdbc</artifactId>
+      <version>2.2.1</version>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.openjpa</groupId>
+      <artifactId>openjpa-xmlstore</artifactId>
+      <version>2.2.1</version>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.openjpa</groupId>
+      <artifactId>openjpa-slice</artifactId>
+      <version>2.2.1</version>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.openjpa</groupId>
+      <artifactId>openjpa-jest</artifactId>
+      <version>2.2.1</version>
       <optional>true</optional>
     </dependency>
     <dependency>
       <groupId>org.eclipse.persistence</groupId>
-      <artifactId>org.eclipse.persistence.asm</artifactId>
+      <artifactId>org.eclipse.persistence.core</artifactId>
       <version>1.0.1</version>
+      <optional>true</optional>
+    </dependency>
+    <dependency>
+      <groupId>asm</groupId>
+      <artifactId>asm</artifactId>
+      <version>3.3.1</version>
       <scope>test</scope>
     </dependency>
     <dependency>
-      <groupId>org.eclipse.persistence</groupId>
-      <artifactId>org.eclipse.persistence.antlr</artifactId>
-      <version>1.0.1</version>
+      <groupId>antlr</groupId>
+      <artifactId>antlr</artifactId>
+      <version>2.7.7</version>
       <scope>test</scope>
     </dependency>
     <dependency>
@@ -77,14 +107,8 @@
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>org.hibernate</groupId>
-      <artifactId>hibernate-cglib-repack</artifactId>
-      <version>2.1_3</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
       <groupId>cglib</groupId>
-      <artifactId>cglib-nodep</artifactId>
+      <artifactId>cglib</artifactId>
       <version>2.2</version>
       <scope>test</scope>
     </dependency>
@@ -101,12 +125,6 @@
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>org.hibernate</groupId>
-      <artifactId>hibernate-annotations</artifactId>
-      <version>3.4.0.GA</version>
-      <optional>true</optional>
-    </dependency>
-    <dependency>
       <groupId>org.hibernate</groupId>
       <artifactId>hibernate-entitymanager</artifactId>
       <version>3.4.0.GA</version>
