--- org.springframework.context.support/pom.xml	2012-02-17 00:33:27.000000000 +0100
+++ org.springframework.context.support/pom.xml-gil	2012-05-26 13:46:06.445609117 +0200
@@ -22,13 +22,13 @@
   </licenses>
   <dependencies>
     <dependency>
-      <groupId>com.bea.commonj</groupId>
-      <artifactId>com.springsource.commonj</artifactId>
+      <groupId>org.apache.geronimo.specs</groupId>
+      <artifactId>geronimo-commonj_1.1_spec</artifactId>
       <version>1.1.0</version>
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>opensymphony</groupId>
+      <groupId>org.quartz-scheduler</groupId>
       <artifactId>quartz</artifactId>
       <version>1.6.2</version>
       <optional>true</optional>
@@ -39,12 +39,12 @@
       <version>1.8.0.10</version>
       <scope>test</scope>
     </dependency>
-    <dependency>
+    <!--dependency>
       <groupId>javax.activation</groupId>
       <artifactId>activation</artifactId>
       <version>1.1</version>
       <scope>provided</scope>
-    </dependency>
+    </dependency-->
     <dependency>
       <groupId>javax.mail</groupId>
       <artifactId>mail</artifactId>
@@ -52,7 +52,7 @@
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>jasperreports</groupId>
+      <groupId>net.sf.jasperreports</groupId>
       <artifactId>jasperreports</artifactId>
       <version>2.0.5</version>
       <optional>true</optional>
