diff --git a/org.springframework.transaction/pom.xml b/org.springframework.transaction/pom.xml
index 01c5263..f748a3f 100644
--- a/org.springframework.transaction/pom.xml
+++ b/org.springframework.transaction/pom.xml
@@ -28,9 +28,9 @@
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>javax.resource</groupId>
-      <artifactId>com.springsource.javax.resource</artifactId>
-      <version>1.5.0</version>
+      <groupId>org.jboss.spec.javax.resource</groupId>
+      <artifactId>jboss-connector-api_1.6_spec</artifactId>
+      <version>1.0.0.Final</version>
       <optional>true</optional>
     </dependency>
     <dependency>
diff --git a/org.springframework.transaction/src/main/java/org/springframework/jca/support/SimpleBootstrapContext.java b/org.springframework.transaction/src/main/java/org/springframework/jca/support/SimpleBootstrapContext.java
index 82d6a23..a5e9ad3 100644
--- a/org.springframework.transaction/src/main/java/org/springframework/jca/support/SimpleBootstrapContext.java
+++ b/org.springframework.transaction/src/main/java/org/springframework/jca/support/SimpleBootstrapContext.java
@@ -21,7 +21,9 @@ import java.util.Timer;
 import javax.resource.spi.BootstrapContext;
 import javax.resource.spi.UnavailableException;
 import javax.resource.spi.XATerminator;
+import javax.resource.spi.work.WorkContext;
 import javax.resource.spi.work.WorkManager;
+import javax.transaction.TransactionSynchronizationRegistry;
 
 /**
  * Simple implementation of the JCA 1.5 {@link javax.resource.spi.BootstrapContext}
@@ -77,4 +79,11 @@ public class SimpleBootstrapContext implements BootstrapContext {
 		return new Timer();
 	}
 
+	public TransactionSynchronizationRegistry getTransactionSynchronizationRegistry() {
+		return null;
+	}
+
+	public boolean isContextSupported(Class<? extends WorkContext> workContextClass) {
+		return true;
+	}
 }
