diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.aop/pom.xml spring-framework-3.1.4.RELEASE-gil/org.springframework.aop/pom.xml
--- spring-framework-3.1.4.RELEASE/org.springframework.aop/pom.xml	2013-10-12 17:44:12.257622183 +0200
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.aop/pom.xml	2013-10-12 17:45:44.207529006 +0200
@@ -52,9 +52,9 @@
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-asm</artifactId>
-      <version>${project.version}</version>
+      <groupId>asm</groupId>
+      <artifactId>asm</artifactId>
+      <version>3.3.1</version>
       <scope>compile</scope>
     </dependency>
     <dependency>
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.context/pom.xml spring-framework-3.1.4.RELEASE-gil/org.springframework.context/pom.xml
--- spring-framework-3.1.4.RELEASE/org.springframework.context/pom.xml	2013-10-12 17:44:12.258622127 +0200
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.context/pom.xml	2013-10-12 17:47:06.693981506 +0200
@@ -121,8 +121,8 @@
     <dependency>
       <groupId>asm</groupId>
       <artifactId>asm</artifactId>
-      <version>3.0</version>
-      <optional>true</optional>
+      <version>3.3.1</version>
+      <scope>compile</scope>
     </dependency>
     <dependency>
       <groupId>joda-time</groupId>
@@ -161,12 +161,6 @@
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-asm</artifactId>
-      <version>${project.version}</version>
-      <scope>compile</scope>
-    </dependency>
-    <dependency>
       <groupId>org.apache.geronimo.specs</groupId>
       <artifactId>geronimo-jta_1.1_spec</artifactId>
       <version>1.1</version>
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.context/src/main/java/org/springframework/scripting/support/ScriptFactoryPostProcessor.java spring-framework-3.1.4.RELEASE-gil/org.springframework.context/src/main/java/org/springframework/scripting/support/ScriptFactoryPostProcessor.java
--- spring-framework-3.1.4.RELEASE/org.springframework.context/src/main/java/org/springframework/scripting/support/ScriptFactoryPostProcessor.java	2013-01-23 14:58:18.000000000 +0100
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.context/src/main/java/org/springframework/scripting/support/ScriptFactoryPostProcessor.java	2013-10-12 17:43:18.592631502 +0200
@@ -20,7 +20,7 @@
 import java.util.Iterator;
 import java.util.Map;
 
-import net.sf.cglib.asm.Type;
+import org.objectweb.asm.Type;
 import net.sf.cglib.core.Signature;
 import net.sf.cglib.proxy.InterfaceMaker;
 import org.apache.commons.logging.Log;
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.core/pom.xml spring-framework-3.1.4.RELEASE-gil/org.springframework.core/pom.xml
--- spring-framework-3.1.4.RELEASE/org.springframework.core/pom.xml	2013-10-12 17:44:12.260622016 +0200
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.core/pom.xml	2013-10-12 17:49:02.330614178 +0200
@@ -28,9 +28,15 @@
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-asm</artifactId>
-      <version>${project.version}</version>
+      <groupId>asm</groupId>
+      <artifactId>asm</artifactId>
+      <version>3.3.1</version>
+      <scope>compile</scope>
+    </dependency>
+    <dependency>
+      <groupId>asm</groupId>
+      <artifactId>asm-commons</artifactId>
+      <version>3.3.1</version>
       <scope>compile</scope>
     </dependency>
     <dependency>
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/LocalVariableTableParameterNameDiscoverer.java spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/LocalVariableTableParameterNameDiscoverer.java
--- spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/LocalVariableTableParameterNameDiscoverer.java	2013-01-23 14:58:18.000000000 +0100
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/LocalVariableTableParameterNameDiscoverer.java	2013-10-12 17:43:18.578632294 +0200
@@ -28,12 +28,12 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
-import org.springframework.asm.ClassReader;
-import org.springframework.asm.Label;
-import org.springframework.asm.MethodVisitor;
-import org.springframework.asm.Opcodes;
-import org.springframework.asm.Type;
-import org.springframework.asm.commons.EmptyVisitor;
+import org.objectweb.asm.ClassReader;
+import org.objectweb.asm.Label;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.commons.EmptyVisitor;
 import org.springframework.util.ClassUtils;
 
 /**
@@ -108,7 +108,7 @@
 		try {
 			ClassReader classReader = new ClassReader(is);
 			Map<Member, String[]> map = new ConcurrentHashMap<Member, String[]>();
-			classReader.accept(new ParameterNameDiscoveringVisitor(clazz, map), false);
+			classReader.accept(new ParameterNameDiscoveringVisitor(clazz, map), 0);
 			return map;
 		}
 		catch (IOException ex) {
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationAttributesReadingVisitor.java spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationAttributesReadingVisitor.java
--- spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationAttributesReadingVisitor.java	2013-01-23 14:58:18.000000000 +0100
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationAttributesReadingVisitor.java	2013-10-12 17:43:18.499636763 +0200
@@ -28,8 +28,8 @@
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
-import org.springframework.asm.AnnotationVisitor;
-import org.springframework.asm.Type;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.Type;
 import org.springframework.core.annotation.AnnotationAttributes;
 import org.springframework.core.annotation.AnnotationUtils;
 import org.springframework.util.ObjectUtils;
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationMetadataReadingVisitor.java spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationMetadataReadingVisitor.java
--- spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationMetadataReadingVisitor.java	2013-01-23 14:58:18.000000000 +0100
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/AnnotationMetadataReadingVisitor.java	2013-10-12 17:43:18.498636820 +0200
@@ -23,9 +23,9 @@
 import java.util.Map;
 import java.util.Set;
 
-import org.springframework.asm.AnnotationVisitor;
-import org.springframework.asm.MethodVisitor;
-import org.springframework.asm.Type;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Type;
 import org.springframework.core.annotation.AnnotationAttributes;
 import org.springframework.core.type.AnnotationMetadata;
 import org.springframework.core.type.MethodMetadata;
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/ClassMetadataReadingVisitor.java spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/ClassMetadataReadingVisitor.java
--- spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/ClassMetadataReadingVisitor.java	2013-01-23 14:58:18.000000000 +0100
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/ClassMetadataReadingVisitor.java	2013-10-12 17:43:18.500636707 +0200
@@ -19,13 +19,13 @@
 import java.util.LinkedHashSet;
 import java.util.Set;
 
-import org.springframework.asm.AnnotationVisitor;
-import org.springframework.asm.Attribute;
-import org.springframework.asm.ClassVisitor;
-import org.springframework.asm.FieldVisitor;
-import org.springframework.asm.MethodVisitor;
-import org.springframework.asm.Opcodes;
-import org.springframework.asm.commons.EmptyVisitor;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.Attribute;
+import org.objectweb.asm.ClassVisitor;
+import org.objectweb.asm.FieldVisitor;
+import org.objectweb.asm.MethodVisitor;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.commons.EmptyVisitor;
 import org.springframework.core.type.ClassMetadata;
 import org.springframework.util.ClassUtils;
 
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/MetadataReader.java spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/MetadataReader.java
--- spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/MetadataReader.java	2013-01-23 14:58:18.000000000 +0100
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/MetadataReader.java	2013-10-12 17:43:18.499636763 +0200
@@ -22,7 +22,7 @@
 
 /**
  * Simple facade for accessing class metadata,
- * as read by an ASM {@link org.springframework.asm.ClassReader}.
+ * as read by an ASM {@link org.objectweb.asm.ClassReader}.
  *
  * @author Juergen Hoeller
  * @since 2.5
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/MethodMetadataReadingVisitor.java spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/MethodMetadataReadingVisitor.java
--- spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/MethodMetadataReadingVisitor.java	2013-01-23 14:58:18.000000000 +0100
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/MethodMetadataReadingVisitor.java	2013-10-12 17:43:18.501636650 +0200
@@ -19,11 +19,11 @@
 import java.util.LinkedHashMap;
 import java.util.Map;
 
-import org.springframework.asm.AnnotationVisitor;
-import org.springframework.asm.MethodAdapter;
-import org.springframework.asm.Opcodes;
-import org.springframework.asm.Type;
-import org.springframework.asm.commons.EmptyVisitor;
+import org.objectweb.asm.AnnotationVisitor;
+import org.objectweb.asm.MethodAdapter;
+import org.objectweb.asm.Opcodes;
+import org.objectweb.asm.Type;
+import org.objectweb.asm.commons.EmptyVisitor;
 import org.springframework.core.annotation.AnnotationAttributes;
 import org.springframework.core.type.MethodMetadata;
 import org.springframework.util.MultiValueMap;
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/SimpleMetadataReaderFactory.java spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/SimpleMetadataReaderFactory.java
--- spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/SimpleMetadataReaderFactory.java	2013-01-23 14:58:18.000000000 +0100
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/SimpleMetadataReaderFactory.java	2013-10-12 17:43:18.500636707 +0200
@@ -25,7 +25,7 @@
 
 /**
  * Simple implementation of the {@link MetadataReaderFactory} interface,
- * creating a new ASM {@link org.springframework.asm.ClassReader} for every request.
+ * creating a new ASM {@link org.objectweb.asm.ClassReader} for every request.
  *
  * @author Juergen Hoeller
  * @since 2.5
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/SimpleMetadataReader.java spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/SimpleMetadataReader.java
--- spring-framework-3.1.4.RELEASE/org.springframework.core/src/main/java/org/springframework/core/type/classreading/SimpleMetadataReader.java	2013-01-23 14:58:18.000000000 +0100
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.core/src/main/java/org/springframework/core/type/classreading/SimpleMetadataReader.java	2013-10-12 17:43:18.583632011 +0200
@@ -20,14 +20,14 @@
 import java.io.IOException;
 import java.io.InputStream;
 
-import org.springframework.asm.ClassReader;
+import org.objectweb.asm.ClassReader;
 import org.springframework.core.io.Resource;
 import org.springframework.core.type.AnnotationMetadata;
 import org.springframework.core.type.ClassMetadata;
 
 /**
  * {@link MetadataReader} implementation based on an ASM
- * {@link org.springframework.asm.ClassReader}.
+ * {@link org.objectweb.asm.ClassReader}.
  *
  * <p>Package-visible in order to allow for repackaging the ASM library
  * without effect on users of the <code>core.type</code> package.
@@ -56,7 +56,7 @@
 		}
 
 		AnnotationMetadataReadingVisitor visitor = new AnnotationMetadataReadingVisitor(classLoader);
-		classReader.accept(visitor, true);
+		classReader.accept(visitor, ClassReader.SKIP_DEBUG);
 		
 		this.annotationMetadata = visitor;
 		// (since AnnotationMetadataReader extends ClassMetadataReadingVisitor)
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.web.portlet/pom.xml spring-framework-3.1.4.RELEASE-gil/org.springframework.web.portlet/pom.xml
--- spring-framework-3.1.4.RELEASE/org.springframework.web.portlet/pom.xml	2013-10-12 17:44:12.297619955 +0200
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.web.portlet/pom.xml	2013-10-12 17:50:28.177888493 +0200
@@ -59,9 +59,9 @@
     </dependency>
     <!-- common logging dependency is in spring-core -->
     <dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-asm</artifactId>
-      <version>${project.version}</version>
+      <groupId>asm</groupId>
+      <artifactId>asm</artifactId>
+      <version>3.3.1</version>
       <scope>compile</scope>
     </dependency>
     <dependency>
diff -Nru spring-framework-3.1.4.RELEASE/org.springframework.web.servlet/pom.xml spring-framework-3.1.4.RELEASE-gil/org.springframework.web.servlet/pom.xml
--- spring-framework-3.1.4.RELEASE/org.springframework.web.servlet/pom.xml	2013-10-12 17:44:12.307619398 +0200
+++ spring-framework-3.1.4.RELEASE-gil/org.springframework.web.servlet/pom.xml	2013-10-12 17:52:23.878519789 +0200
@@ -158,9 +158,9 @@
       <optional>true</optional>
     </dependency>
     <dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-asm</artifactId>
-      <version>${project.version}</version>
+      <groupId>asm</groupId>
+      <artifactId>asm</artifactId>
+      <version>3.3.1</version>
       <scope>compile</scope>
     </dependency>
     <dependency>
